gson/src/main/java/com/google/gson/internal/reflect/ReflectionHelper.java

```diff
@@ -169,22 +177,34 @@ public <T> Constructor<T> getCanonicalRecordConstructor(Class<T> raw) {
       try {
         Object recordComponents = getRecordComponents.invoke(raw);
         int componentCount = Array.getLength(recordComponents);
-        Class[] recordFields = new Class[componentCount];
+        Class<?>[] recordComponentType = new Class<?>[componentCount];
         for (int i = 0; i < componentCount; i++) {
-          recordFields[i] =
-                  Class.class.cast(getType.invoke(Array.get(recordComponents, i)));
+          recordComponentType[i] = Class.class.cast(getType.invoke(Array.get(recordComponents, i)));
         }
-        return raw.getConstructor(recordFields);
-      } catch (IllegalAccessException e) {
-        throw createExceptionForUnexpectedIllegalAccess(e);
+        return raw.getDeclaredConstructor(recordComponentType);
       } catch (ReflectiveOperationException e) {
-        throw new RuntimeException("Failed to reflect into record components on"
-                + " class [" + raw + "], this is unexpected behaviour, as these methods should"
-                + " not throw when they are defined on Class",
-                e
-        );
+        throw createExceptionForRecordReflectionException(e);
       }
     }
+
+    @Override
+    public Method getAccessor(Class<?> raw, Field field) {
+      try {
+        Object recordComponents = getRecordComponents.invoke(raw);
+        int componentCount = Array.getLength(recordComponents);
+        for (int i = 0; i < componentCount; i++) {
+          Object recordComponent = Array.get(recordComponents, i);
+          String componentName = String.class.cast(getName.invoke(recordComponent));
+          if (componentName.equals(field.getName())) {
+            return Method.class.cast(getAccessor.invoke(recordComponent));
```

Comment:
```suggestion
          String componentName = (String) getName.invoke(recordComponent);
          if (componentName.equals(field.getName())) {
            return (Method) getAccessor.invoke(recordComponent);
```

Is there are a reason why you are writing so often `...class.cast(...)` here? I don't think it provides any advantage over regular casts and is a bit more verbose.

Reply:
Agree, I guess I was just in a mode to use Reflection for everything :)